aws/data_source_aws_eks_cluster_auth.go

```diff
@@ -0,0 +1,49 @@
+package aws
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/kubernetes-sigs/aws-iam-authenticator/pkg/token"
+)
+
+func dataSourceAwsEksClusterAuth() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsEksClusterAuthRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
```

Comment:
Nit: `ForceNew: true` is redundant for data sources as they will always refresh.

```suggestion

```